### DISCUSSION TITLE
How to read weight of edges to calculate "length" of paths correctly?

### DISCUSSION THREAD

**Discussion by forestbat** [networkx/networkx#5915] [Q&A] [answered]
I wrote a graph like this code:
`origin_graph.add_edge(src, dest, weight=line.length)`
And I want to compare length of paths each other：
`if nx.shortest_path_length(origin_graph, source, target, weight='weight') < nx.dag_longest_path(origin_graph, source, target, weight='weight')`
Then the process crashed and told me：

> File "……", line 294, in ……
    nx.dag_longest_path(origin_graph, source, target, weight='weight'):
  File "C:\ProgramData\Anaconda3\envs\……\lib\site-packages\networkx\utils\decorators.py", line 845, in func
    return argmap._lazy_compile(__wrapper)(*args, **kwargs)
TypeError: argmap_dag_longest_path_6() got multiple values for argument 'weight'

What happened and how to solve it？

**Comment by MridulS** [comment, marked as answer]
The function `nx.dag_longest_path` returns the longest path in the directed acyclic graph. You can not choose a source and target. In the example provided by doing `nx.dag_longest_path(origin_graph, source, target, weight='weight')` you are passing in the `weight` twice, as the function expects the second argument to be the `weight`. You can have a look at the documentation at https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.dag.dag_longest_path.html 


The line should be changed to `if nx.shortest_path_length(origin_graph, source, target, weight='weight') < nx.dag_longest_path(origin_graph, weight='weight')`. I'm not sure what you are comparing, if you just want the longest path in the directed acyclic graph then this should work.

**Reply by forestbat** [reply]
Oh，I'm too rough and stupid，the correct answer is `if nx.shortest_path_length(origin_graph, source, target, weight='weight') < nx.dag_longest_path_length(origin_graph, weight='weight')`
